tests/topotests/isis_srv6_te_topo1/test_isis_srv6_te_topo1.py

```diff
@@ -0,0 +1,472 @@
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib import topotest
+from lib.topogen import Topogen, TopoRouter, get_topogen
+from lib.topolog import logger
+from lib.common_config import (
+    required_linux_kernel_version,
+    create_interface_in_kernel,
+)
+
+pytestmark = [pytest.mark.isisd, pytest.mark.bgpd]
+
+
+def build_topo(tgen):
+    """Build function"""
+
+    # Define FRR Routers
+    tgen.add_router("rt1")
+    tgen.add_router("rt2")
+    tgen.add_router("rt3")
+    tgen.add_router("rt4")
+    tgen.add_router("rt5")
+    tgen.add_router("rt6")
+    tgen.add_router("dst")
+    tgen.add_router("src")
+
+    # Define connections
+    switch = tgen.add_switch("s1")
+    switch.add_link(tgen.gears["rt1"], nodeif="eth-sw1")
+    switch.add_link(tgen.gears["rt2"], nodeif="eth-sw1")
+    switch.add_link(tgen.gears["rt3"], nodeif="eth-sw1")
+
+    switch = tgen.add_switch("s2")
+    switch.add_link(tgen.gears["rt2"], nodeif="eth-rt4-1")
+    switch.add_link(tgen.gears["rt4"], nodeif="eth-rt2-1")
+
+    switch = tgen.add_switch("s3")
+    switch.add_link(tgen.gears["rt2"], nodeif="eth-rt4-2")
+    switch.add_link(tgen.gears["rt4"], nodeif="eth-rt2-2")
+
+    switch = tgen.add_switch("s4")
+    switch.add_link(tgen.gears["rt3"], nodeif="eth-rt5-1")
+    switch.add_link(tgen.gears["rt5"], nodeif="eth-rt3-1")
+
+    switch = tgen.add_switch("s5")
+    switch.add_link(tgen.gears["rt3"], nodeif="eth-rt5-2")
+    switch.add_link(tgen.gears["rt5"], nodeif="eth-rt3-2")
+
+    switch = tgen.add_switch("s6")
+    switch.add_link(tgen.gears["rt4"], nodeif="eth-rt5")
+    switch.add_link(tgen.gears["rt5"], nodeif="eth-rt4")
+
+    switch = tgen.add_switch("s7")
+    switch.add_link(tgen.gears["rt4"], nodeif="eth-rt6")
+    switch.add_link(tgen.gears["rt6"], nodeif="eth-rt4")
+
+    switch = tgen.add_switch("s8")
+    switch.add_link(tgen.gears["rt5"], nodeif="eth-rt6")
+    switch.add_link(tgen.gears["rt6"], nodeif="eth-rt5")
+
+    switch = tgen.add_switch("s9")
+    switch.add_link(tgen.gears["rt6"], nodeif="eth-dst")
+    switch.add_link(tgen.gears["dst"], nodeif="eth-rt6")
+
+    switch = tgen.add_switch("s10")
+    switch.add_link(tgen.gears["rt1"], nodeif="eth-src")
+    switch.add_link(tgen.gears["src"], nodeif="eth-rt1")
+
+    # Add dummy interface for SRv6
+    create_interface_in_kernel(
+        tgen,
+        "rt1",
+        "sr0",
+        "2001:db8::1",
+        netmask="128",
+        create=True,
+    )
+    create_interface_in_kernel(
+        tgen,
+        "rt2",
+        "sr0",
+        "2001:db8::2",
+        netmask="128",
+        create=True,
+    )
+    create_interface_in_kernel(
+        tgen,
+        "rt3",
+        "sr0",
+        "2001:db8::3",
+        netmask="128",
+        create=True,
+    )
+    create_interface_in_kernel(
+        tgen,
+        "rt4",
+        "sr0",
+        "2001:db8::4",
+        netmask="128",
+        create=True,
+    )
+    create_interface_in_kernel(
```

Comment:
this would be nice to really have the sr0 interface active.
for instance, use the following call for each device.
```
    shutdown_bringup_interface_in_kernel(
        tgen,
        "rt2",
        "sr0",
        ifaceaction=True
    )
```